File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        mini=prices[0]
        maxProfit=0
        for i in range(len(prices)):
            maxProfit=max(maxProfit,prices[i]-mini)
            mini=min(mini,prices[i])

        return maxProfit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini=prices[0];
        int maxProfit=0;
        for(int i=0; i<prices.length; i++){
            maxProfit=Math.max(maxProfit,prices[i]-mini);
            mini=Math.min(mini,prices[i]);
        }

        return maxProfit;
    }
}